src/prefect/client/orchestration.py

```diff
@@ -3143,16 +3154,24 @@ async def read_automations(self) -> List[Automation]:
         response.raise_for_status()
         return pydantic.parse_obj_as(List[Automation], response.json())
 
-    async def find_automation(self, id_or_name: str) -> Optional[Automation]:
-        try:
-            id = UUID(id_or_name)
-        except ValueError:
-            id = None
+    async def find_automation(
+        self, id_or_name: Union[str, UUID], exit_if_not_found: bool = True
```

Comment:
```suggestion
        self, id_or_name: Union[str, UUID]
```

This extra `kwarg` was something I mistakenly left in the first version and then removed later.  We don't need it